mindsdb_native/libs/helpers/text_helpers.py

```diff
@@ -170,103 +168,58 @@ def isascii(string):
+    if all_same_length and len(data) == len(set(data)):
+        str_data = [str(x) for x in data]
+
+        randomness_per_index = []
+        for i, _ in enumerate(str_data[0]):
+            N = len(set(x[i] for x in str_data))
+            S = st.entropy([*Counter(x[i] for x in str_data).values()])
+            randomness_per_index.append(S / np.log(N))
+
+        if np.mean(randomness_per_index) > 0.95:
+            return 'Hash-like identifier'
+
+    # Detect foreign key
+    if data_subtype == DATA_SUBTYPES.INT:
+        if _is_foreign_key_name(column_name):
+            return 'Foregin key'
+
+    if _is_identifier_name(column_name) or data_type == DATA_TYPES.CATEGORICAL:
+        if unquie_pct > 0.98:
+            if is_uuid:
+                return 'UUID'
             else:
-                if (x - prev) != 1:
-                    break
-                else:
-                    prev = x
-        else:
-            return 'Auto-incrementing Identifier'
-
-    # Detect UUID
-    if foregin_key_type:
-        is_uuid = False
-    else:
-        all_same_length = all(len(str(data[0])) == len(str(x)) for x in data)
-        uuid_charset = set('0123456789abcdefABCDEF-')
-        all_uuid_charset = all(set(str(x)).issubset(uuid_charset) for x in data)
-        is_uuid = all_uuid_charset and all_same_length
-
-
-        # If all data points are strings of equal length
-        # then compute entropy per each index through all data
-        #
-        # Example:
-        #
-        #   column
-        # 1 'wqk5'
-        # 2 'wq6z'
-        # 3 'wqv7'
-        # 4 'eq8O'
-        # 5 'eqkO'
-        # 6 'eqyS'
-        # 7 'eqAe' 
-        #    ||||
-        #    ||||-------------------- index 3
-        #    |||                      Counter({5: 1, z: 1, 7: 1, O: 2, s: 1, e: 1})
-        #    |||                      S = entropy[1, 1, 1, 2, 1, 1]
-        #    |||                      randomness = S / np.log(6) <----- 6 unique values at this index
-        #    |||
-        #    |||--------------------- index 2
-        #    ||                       Counter({k: 2, 6: 1, v: 1, 8: 1, Y: 1, A: 1})
-        #    ||                       S = entropy[2, 1, 1, 1, 1, 1]
-        #    ||                       randomness = S / np.log(6) <----- 6 unique values at this index
-        #    ||
-        #    ||---------------------- index 1
-        #    |                        Counter({q: 7})
-        #    |                        S = entropy[7]
-        #    |                        randomness = S / np.log(1) <----- 1 unique value at this index
-        #    |
-        #    |----------------------- index 0
-        #                             Counter({w: 3, e: 4})
-        #                             S = entropy[3, 4]
-        #                             randomness = S / np.log(2) <----- 2 unique values at this index
-        #
-        # Scaling entropy by np.log(num_of_unique_values) produces a number in range [0, 1]
```

Comment:
why is the comment deleted?

Reply:
I thought this approach was no longer needed, so no point to leave the comment in. Or is there still a situation when we'd want to reintroduce this ?

Reply:
but the code for it is still there: https://github.com/mindsdb/mindsdb_native/pull/200/files#diff-d26ce10f25a7c27ed6efdd7906661481R211

Reply:
added back in, my fault.